### ISSUE TITLE
clang-tidy: false positive from `hicpp-noexcept-move` for templated, defaulted move constructors

### ISSUE THREAD

**Issue by AMS21** [llvm/llvm-project#56026]
Given the following source code:
```cpp
#include <type_traits>

struct A {
    A(A&&) = default;
};

template <typename T1>
struct B {
    B(B&&) = default;
};

static_assert(std::is_nothrow_move_constructible<A>::value, "");
static_assert(std::is_nothrow_move_constructible<B<int>>::value, "");
```

Running `clang-tidy -checks=hicpp-noexcept-move`
We get the following output:
```
<source>:9:5: warning: move constructors should be marked noexcept [hicpp-noexcept-move]
    B(B&&) = default;
    ^
            noexcept 
1 warning generated.
```

Although the move constructor is automatically noexcept due to it being defaulted just like struct `A` and otherwise the static_assert would fail on line 13.

[godbolt link](https://godbolt.org/z/nnjrTPqhj)

**Comment by AMS21** [comment]
Same is true for templated defaulted move assignment operators
[godbolt](https://godbolt.org/z/MsczWM137)
